# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1232650773 18000
#      Thu Jan 22 13:59:33 2009 -0500
# Node ID 83b8c739d62612c96809a612511e8af2116626ed
# Parent  da61d0f7ce0b7e3578652b58057883003707bd03
toplev.cc: check OCTAVE_USE_OS_X_API instead of __APPLE__ && __MACH__

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,13 @@
 2009-01-22  John W. Eaton  <jwe@octave.org>
 
+	* toplev.cc (octave_config_info): Check OCTAVEUSE_OS_X_API instead
+	of __APPLE__ && __MACH__.
+
 	* display.cc (display_info::init): Get info for Windows and OS X
 	systems.
 
 2009-01-22  Jaroslav Hajek  <highegg@gmail.com>
 
 	* pt-idx.cc (tree_index_expression::lvalue): Correct tmpi when seeing
 	a plain struct component.
 	* ov-struct.cc (octave_struct::subsasgn): Use proper type substr in
diff --git a/src/toplev.cc b/src/toplev.cc
--- a/src/toplev.cc
+++ b/src/toplev.cc
@@ -1278,17 +1278,17 @@ specified option.\n\
 
 #if defined (WIN32)
       windows_system = true;
 #if !defined (__CYGWIN__)
       unix_system = false;
 #endif
 #endif
 
-#if defined (__APPLE__) && defined (__MACH__)
+#if defined (OCTAVE_USE_OS_X_API)
       mac_system = true;
 #endif
 
       m.assign ("unix", octave_value (unix_system));
       m.assign ("mac", octave_value (mac_system));
       m.assign ("windows", octave_value (windows_system));
 
       initialized = true;
